docs/add-ons/portworx.md

```diff
@@ -0,0 +1,142 @@
+# Portworx add-on for EKS Blueprints
+
+## Introduction
+
+[Portworx](https://portworx.com/) is a Kubernetes data services platform that provides persistent storage, data protection, disaster recovery, and other capabilities for containerized applications. This blueprint installs Portworx on Amazon Elastic Kubernetes Service (EKS) environment.
+
+- [Helm chart](https://github.com/portworx/helm)
+
+## Examples Blueprint
+
+To get started look at these sample [blueprints](https://github.com/portworx/terraform-eksblueprints-portworx-addon/tree/main/blueprint).
+
+## Requirements
+
+For the add-on to work, Portworx needs additional permission to AWS resources which can be provided in the following two ways. The different flows are also covered in [sample blueprints](https://github.com/portworx/terraform-eksblueprints-portworx-addon/tree/main/blueprint):
+
+## Method 1: Custom IAM policy
+
+1. Add the below code block in your terraform script to create a policy with the required permissions. Make a note of the resource name for the policy you created:
+
+```
+resource "aws_iam_policy" "<policy-resource-name>" {
+  name = "<policy-name>"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "ec2:AttachVolume",
+          "ec2:ModifyVolume",
+          "ec2:DetachVolume",
+          "ec2:CreateTags",
+          "ec2:CreateVolume",
+          "ec2:DeleteTags",
+          "ec2:DeleteVolume",
+          "ec2:DescribeTags",
+          "ec2:DescribeVolumeAttribute",
+          "ec2:DescribeVolumesModifications",
+          "ec2:DescribeVolumeStatus",
+          "ec2:DescribeVolumes",
+          "ec2:DescribeInstances",
+          "autoscaling:DescribeAutoScalingGroups"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+    ]
+  })
+}
+```
+
+2. Run `terraform apply` command for the policy (replace it with your resource name):
+
+```bash
+terraform apply -target="aws_iam_policy.<policy-resource-name>"
+```
+3. Attach the newly created AWS policy ARN to the node groups in your cluster:
+
+```
+ managed_node_groups = {
+    node_group_1 = {
+      node_group_name           = "my_node_group_1"
+      instance_types            = ["t2.small"]
+      min_size                  = 3
+      max_size                  = 3
+      subnet_ids                = module.vpc.private_subnets
+
+      #Add this line to the code block or add the new policy ARN to the list if it already exists
+      additional_iam_policies   = [aws_iam_policy.<policy-resource-name>.arn]
```

Comment:
Why not use IRSA?

Reply:
Portworx needs those instance level privileges and isn't designed to get the same permission through service accounts and that is why we need to resort to using additional IAM policies or AWS credentials